src/DIRAC/FrameworkSystem/API/AuthHandler.py

```diff
@@ -382,7 +382,7 @@ def get_redirect(self, state, error=None, error_description="", chooseScope=[]):
 
         # RESPONSE to basic DIRAC client request
         resp = self.server.create_authorization_response(response, grant_user)
-        if not resp.payload.startswith("<!DOCTYPE html>"):
+        if isinstance(resp.payload, str) and not resp.payload.startswith("<!DOCTYPE html>"):
```

Comment:
For my info, what type can `resp.payload` be? bytes?

Reply:
Theoretically is possible

Reply:
The code expects to have `dict` or `str` here.

Reply:
In which case would it be bytes? And don't you want to decode in that case?

Reply:
> In which case would it be bytes? 

It depends from `authlib` library, first of all, because it also uses the `handle_response` method, currently in my development I do not pass bytes there.

> And don't you want to decode in that case?

But even if bytes are passed then the tornado [finish](https://www.tornadoweb.org/en/stable/web.html#tornado.web.RequestHandler.finish) method will be able to convert them